### PR TITLE
PUB-90 Add support for gzip compression/decompression 

### PR DESCRIPTION
**Pull request by rndD**
* Pako package for gzip support
* Topic option to enable compression for publish messages
* Message now can auto check if it's compressed and decompresses on `toJson()`. Added test to cover this.
* Added examples to test all this

### CODE REVIEW

File: src/interface/publishOptions.ts [hunk 1]
```diff
@@ -58,6 +63,12 @@ export interface TopicPublishOptions {
 export interface PublishOptions {
   retryConfig?: RetryConfig;
   attributes?: Attributes;
+
+  // TODO why I have to copy that?
```
Comment (rndD) [inline src/interface/publishOptions.ts:67]:
```suggestion
```

File: src/topic/index.ts [hunk 2]
```diff
@@ -27,11 +27,13 @@ type PayloadInput<P extends Payload> = Omit<P, keyof Payload>;
 export interface TopicOptions {
   addTimeStamp?: boolean;
   retryConfig?: RetryConfig;
+  enableGZipCompression?: boolean;
 }
 
 export interface TopicProperties {
   topicName: string;
   project?: GooglePubSubProject;
+  enableGZipCompression?: boolean;
```
Comment (rndD) [inline src/topic/index.ts:36]: I felt strange copy-pasting this line over and over across the project...

File: src/message/index.ts [hunk 3]
```diff
@@ -29,6 +28,14 @@ export default class Message<T = unknown> {
     return instance;
   }
 
+  public toJSON(): T {
+    let data = this.data;
+    if (isGzipCompressed(data)) {
+      data = this.decompress();
+    }
+    return JSON.parse(data.toString());
+  }
```
Comment (rohit-gohri) [inline src/message/index.ts:37]: Nice!

File: src/topic/index.ts (hunk 2 above)
Comment (rohit-gohri) [reply src/topic/index.ts:36]: Yeah, there is a lot of duplication across interfaces

Comment (rohit-gohri) [inline src/topic/index.ts:36]: But I don't think it needs to be added in this particular place as these are the static properties of the topic

File: src/message/gzip.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const isGzipCompressed = (buf: Buffer): boolean => {
```
Comment (rndD) [inline src/message/gzip.ts:1]: add a comment with the link to docs

File: src/message/index.ts (hunk 3 above)
Comment (rndD) [reply src/message/index.ts:37]: I need to add this behavior to docs.

**Review by rohit-gohri — APPROVED** [review]
LGTM

File: src/message/gzip.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
+// @see https://en.wikipedia.org/wiki/Gzip#:~:text=%22gzip%22%20is%20often%20also%20used,and%20the%20operating%20system%20ID.
```
[the remark this answers is not in the document]
Comment (rndD) [reply src/message/gzip.ts:1]: Probably it's the  'highlight text' feature is Chrome only. I will just copy the text.

**Review by rohit-gohri — APPROVED** [review]

**Pull request merged**